xonsh/commands_cache.py

```diff
@@ -133,7 +133,10 @@ def lazy_locate_binary(self, name):
                 if os.path.isfile(full_name)
             ), None)
             if local_bin:
-                return os.path.abspath(os.path.relpath(local_bin, cwd))
+                if os.path.splitdrive(cwd)[0] != os.path.splitdrive(local_bin)[0]:  # if cwd not on same drive as bin
+                    return os.path.abspath(local_bin)                               # avoid ValueError in relpath()
+                else:
+                    return os.path.abspath(os.path.relpath(local_bin, cwd))
```

Comment:
@bobhy I tried to look at this in more detail to clean it up further, and I don't understand why cwd and the binary can ever be on different drives here. 

We should only be here if local_bin is a file in the current working directory. Right?